src/libraries/PCPricing.sol

```diff
@@ -0,0 +1,303 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+/// @title Preferred Currency Pricing (PCPricing) Library
+/// @author Permissionless Games & ChatGpt
+/// @notice This library enables dynamic price adjustments based on preferred currency usage.
+/// @dev When the anchor currency is used, the price of all other tracked currencies can be decreased.
+///      A universal adjustment factor applies to all non-anchor currencies, promoting dynamic pricing models.
+
+library PCPricing {
+    using PCPricing for PCPricing.PricingData;
+    /// @notice Emitted when a new anchor currency is set
+    /// @param currency The currency that was set as the anchor
+    /// @param price The price of the anchor currency
+
+    event AnchorCurrencySet(bytes indexed currency, uint256 price);
+
+    /// @notice Emitted when the adjustment factor is updated
+    /// @param numerator The numerator of the adjustment factor
+    /// @param denominator The denominator of the adjustment factor
+    event AdjustmentFactorSet(uint256 numerator, uint256 denominator);
+
+    /// @notice Emitted when a new currency price is set
+    /// @param currency The currency that was set
+    /// @param price The price of the currency
+    event CurrencyPriceSet(bytes indexed currency, uint256 price);
+
+    /// @notice Emitted when a currency price is adjusted
+    /// @param currency The currency that was adjusted
+    /// @param newPrice The new price of the currency
+    /// @param increased Whether the price was increased or decreased
+    event CurrencyPriceAdjusted(
+        bytes indexed currency,
+        uint256 newPrice,
+        bool increased
+    );
+
+    /// @notice Emitted when a currency is removed
+    /// @param currency The currency that was removed
+    event CurrencyRemoved(bytes indexed currency);
```

Comment:
What does it mean that a currency gets removed? Does it mean that the contract using the library no longer accepts that currency? These kinds of details definitely belong in the comments.